scapy/contrib/automotive/uds_scan.py

```diff
@@ -594,6 +772,51 @@ def _get_table_entry(tup):
                 label)
 
 
+class UDS_RCStartEnumerator(UDS_RCEnumerator):
+    _description = "Available RoutineControls and negative response per state"
+
+    def _get_initial_requests(self, **kwargs):
+        # type: (Any) -> Iterable[Packet]
+        if "type_list" in kwargs:
+            raise Exception("'type_list' already set in kwargs.")
+        if "scan_range" in kwargs:
+            raise Exception("'scan_range' set in kwargs.")
+        kwargs["type_list"] = [1]
+        return \
+            super(UDS_RCStartEnumerator, self)._get_initial_requests(**kwargs)
+
+
+class UDS_RCSelectiveEnumerator(StagedAutomotiveTestCase):
+    _left_right_width = 132
+
+    @staticmethod
+    def points_to_ranges(points_of_interest, range_size=_left_right_width):
+        # type: (Iterable[int], int) -> Iterable[int]
+        to_scan = []
+        for identifier in points_of_interest:
+            start = max(identifier - range_size, 0)
+            end = min(identifier + range_size + 1, 0x10000)
+            to_scan += list(range(start, end))
+        return sorted(set(to_scan))
```

Comment:
why again `set()` here. This will always expand the list of identifiers. If you want to make sure, all identifiers are unique, than do ```for identifier in list(set(points_of_intrest))``

Reply:
Because point of interests might be close to each other.
For example, POIs: [5, 20]
From the points ranges are calculated:
5: 0-105
20: 0-120

Without unifying it, 0-105 would be scanned twice.

Reply:
ah ok..